src/cache.c

```diff
@@ -1,138 +1,164 @@
+#include "cache.h"
+#include "hashtable.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include "hashtable.h"
-#include "cache.h"
 
 /**
  * Allocate a cache entry
  */
-struct cache_entry *alloc_entry(char *path, char *content_type, void *content, int content_length)
-{
-    ///////////////////
-    // IMPLEMENT ME! //
-    ///////////////////
+struct cache_entry *alloc_entry(char *path, char *content_type, void *content,
+                                int content_length) {
+  ///////////////////
+  // IMPLEMENT ME! //
+  ///////////////////
+
+  struct cache_entry *new_entry = malloc(sizeof(struct cache_entry));
+  new_entry->path = path;
+  new_entry->content_type = content_type;
+  new_entry->content_length = content_length;
+  new_entry->content = content;
```

Comment:
You could improve your code here to the code that we wrote today in the GP session.